### ISSUE TITLE
data shape problem

### ISSUE THREAD

**Issue by kennethliukai** [kuza55/keras-extras#8]
Hi kuza,

I tried the example posted by you in https://medium.com/@kuza55/transparent-multi-gpu-training-on-tensorflow-with-keras-8b0016fd9012#.oj738zfd4

but get the following errors. Could you please show me what's the reason of the error? thanks

Traceback (most recent call last):
  File "train-kai.py", line 135, in <module>
    t()
  File "train-kai.py", line 72, in t
    model = make_parallel(model, 4)
  File "train-kai.py", line 44, in make_parallel
    slice_n = Lambda(get_slice, output_shape=input_shape, arguments={'idx':i,'parts':gpu_count})(x)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 572, in __call__
    self.add_inbound_node(inbound_layers, node_indices, tensor_indices)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 635, in add_inbound_node
    Node.create_node(self, inbound_layers, node_indices, tensor_indices)
  File "/usr/local/lib/python2.7/dist-packages/keras/engine/topology.py", line 166, in create_node
    output_tensors = to_list(outbound_layer.call(input_tensors[0], mask=input_masks[0]))
  File "/usr/local/lib/python2.7/dist-packages/keras/layers/core.py", line 641, in call
    return self.function(x, **arguments)
  File "train-kai.py", line 23, in get_slice
    size = tf.concat(0, [ shape[:1] // parts, shape[1:] ])
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/array_ops.py", line 1030, in concat
    ).assert_is_compatible_with(tensor_shape.scalar())
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/tensor_shape.py", line 735, in assert_is_compatible_with
    raise ValueError("Shapes %s and %s are incompatible" % (self, other))
ValueError: Shapes (2, 1) and () are incompatible

**Comment by mongoose54** [comment]
@kuza55 Same issue here. @kennethliukai I am wondering if there has been an update to Keras/Tensorflow since the post and it has broken the code.

**Comment by bentaylordata** [comment]
Same error, trying to troubleshoot: 
`ValueError: Shapes (2, 1) and () are incompatible`

**Comment by Caduceus96** [comment]
I think you need to change these lines:

```
size = tf.concat(0,[ shape[:1] // parts, shape[1:] ])
stride = tf.concat(0,[ shape[:1] // parts, shape[1:]*0 ])
```

to:

```
size = tf.concat([ shape[:1] // parts, shape[1:] ],axis=0)
stride = tf.concat([ shape[:1] // parts, shape[1:]*0 ],axis=0)
```

**Comment by beeva-enriqueotero** [comment]
It also worked for me :) I think this patch should be merged to master.

Thanks!

**Comment by sumethy** [comment]
This fix also worked for me on Keras 2.0.3 and tf 1.0.1.
